### PR TITLE
Adds support for retaining back up archives as binary as well as text. Closes #1970

### LINKED ISSUE #1970: Retain the back archive from pg_dump with appropriate naming.

### PR DESCRIPTION
**Pull request by binaek**

### CODE REVIEW

File: db/db_local/backup.go [hunk 1]
```diff
@@ -271,8 +272,8 @@ func restoreBackup(ctx context.Context) error {
 		utils.ShowWarning("Could not REFRESH Materialized Views while restoring data. Please REFRESH manually.")
 	}
 
-	if err := os.Remove(databaseBackupFilePath()); err != nil {
-		log.Printf("[WARN] Could not remove Backup data at %s.", databaseBackupFilePath())
+	if err := retainBackup(ctx); err != nil {
+		log.Printf("[WARN] Could not retain backup %s.", databaseBackupFilePath())
```
Comment (kaidaguerre) [inline db/db_local/backup.go:276]: utils.ShowWarning?

"Failed to save database backup file"

include error?

File: db/db_local/backup.go [hunk 2]
```diff
@@ -436,3 +437,38 @@ func updateDynamicLibPath(ctx context.Context, cmd *exec.Cmd) {
...
+// text:   'database-yyyy-MM-dd-hh-mm-ss.sql'
+func retainBackup(ctx context.Context) error {
+	now := time.Now()
+	backupBaseFileName := fmt.Sprintf(
+		"database-%s",
+		now.Format("2006-01-02-15-04-05"),
+	)
+	binaryBackupRetentionFileName := fmt.Sprintf("%s.dump", backupBaseFileName)
+	textBackupRetentionFileName := fmt.Sprintf("%s.sql", backupBaseFileName)
+
```
Comment (kaidaguerre) [inline db/db_local/backup.go:453]:
```
backupDir := filepaths.EnsureBackupsDir()
binaryBackupFilePath := filepath.Join(backupDir, binaryBackupRetentionFileName)
textBackupFilePath := filepath.Join(backupDir, textBackupRetentionFileName)
```

File: db/db_local/backup.go [hunk 3]
```diff
@@ -436,3 +437,38 @@ func updateDynamicLibPath(ctx context.Context, cmd *exec.Cmd) {
...
+	binaryBackupFilePath := filepath.Join(filepaths.EnsureBackupsDir(), binaryBackupRetentionFileName)
+	textBackupFilePath := filepath.Join(filepaths.EnsureBackupsDir(), textBackupRetentionFileName)
+
+	log.Println("[TRACE] moving back up to", binaryBackupFilePath)
```
Comment (kaidaguerre) [inline db/db_local/backup.go:457]: `moving database back up to`

File: db/db_local/backup.go [hunk 4]
```diff
@@ -436,3 +437,38 @@ func updateDynamicLibPath(ctx context.Context, cmd *exec.Cmd) {
...
+	if err := os.Rename(databaseBackupFilePath(), binaryBackupFilePath); err != nil {
+		return err
+	}
+	log.Println("[TRACE] converting back up to", textBackupFilePath)
```
Comment (kaidaguerre) [inline db/db_local/backup.go:461]: `converting database back up to`

**Review by kaidaguerre — CHANGES_REQUESTED** [review]

File: db/db_local/backup.go (hunk 1 above)
Comment (binaek) [reply db/db_local/backup.go:277]: done

File: db/db_local/backup.go (hunk 2 above)
Comment (binaek) [reply db/db_local/backup.go:398]: done

File: db/db_local/backup.go (hunk 3 above)
Comment (binaek) [reply db/db_local/backup.go]: done

File: db/db_local/backup.go (hunk 4 above)
Comment (binaek) [reply db/db_local/backup.go]: done

File: db/db_local/backup.go (hunk 1 above)
Comment (kaidaguerre) [reply db/db_local/backup.go:277]: so you can remove the log line, right?
```
utils.ShowWarning(fmt.Sprintf("Failed to save backup file %s: %v", databaseBackupFilePath(), err))
```

**Review by kaidaguerre — CHANGES_REQUESTED** [review]

File: db/db_local/backup.go [hunk 5]
```diff
@@ -381,6 +382,42 @@ func getTableOfContentsFromBackup(ctx context.Context) ([]string, error) {
...
+	)
+	binaryBackupRetentionFileName := fmt.Sprintf("%s.dump", backupBaseFileName)
+	textBackupRetentionFileName := fmt.Sprintf("%s.sql", backupBaseFileName)
+
+	backupDir := filepaths.EnsureBackupsDir()
+	binaryBackupFilePath := filepath.Join(backupDir, binaryBackupRetentionFileName)
+	textBackupFilePath := filepath.Join(backupDir, textBackupRetentionFileName)
+
+	log.Println("[TRACE] moving database back up to", binaryBackupFilePath)
+	if err := os.Rename(databaseBackupFilePath(), binaryBackupFilePath); err != nil {
```
Comment (binaek) [inline db/db_local/backup.go:403]: is it worth doing an `io.Copy` instead of `os.Rename` here? `os.Rename` cannot work across partitions.

Comment (kaidaguerre) [reply db/db_local/backup.go:403]: may be worth it yes

**Pull request merged**